### PR TITLE
Restricting usage of owner<T> to pointer types

### PR DESCRIPTION
**Pull request by bashrc-real**
Also added basic test for a simple case for integer

### CODE REVIEW

**Comment by bashrc-real** [comment]
Fix for issue #477

**Comment by neilmacintosh** [comment]
Thanks for the PR @bashrc-real! This contains two unrelated changes, however. I'd prefer to separate out the changes to `owner<T>` and `not_null<T>` so that they are separate. Would that be possible?

**Comment by bashrc-real** [comment]
@neilmacintosh  I have seperated the PRs for the two issues. This one only addresses #477 
Seems like the failure in tests is due to platform issue unrelated to this change.

**Comment by neilmacintosh** [comment]
Thanks @bashrc-real, I span off a new build and it did indeed pass. I have a minor comment for you to consider.

**Comment by bashrc-real** [comment]
@neilmacintosh  I dont see your comment in the review. Am I missing something

**Comment by neilmacintosh** [comment]
@bashrc-real sorry, that's probably my fault...it appears I forgot to click "Complete Review" so that it became visible to you. I have done so now!

File: include/gsl/gsl [hunk 1]
```diff
@@ -53,7 +54,16 @@ namespace gsl
...
+//
+// owner
+//
+//
+// owner<T> is designed as a bridge for code that must deal directly with owning pointers for some reason
+//
+// T must be a pointer type
+// - disallow construction from any type other than pointer type
+//
+template <class T, typename = std::enable_if_t<std::is_pointer<T>::value>>
```
Comment (bashrc-real) [inline include/gsl/gsl]: Fixed the NIT. Couldn't understand the comment about ::value?

**Comment by neilmacintosh** [comment]
Thanks @bashrc-real this all looks good to me now!

**Pull request merged**